vault/expiration.go

```diff
@@ -943,6 +956,27 @@ func (m *ExpirationManager) lazyRevokeInternal(ctx context.Context, leaseID stri
 	return nil
 }
 
+// TODO should context passed to revokeCommon contain the lease namespace?
+// should be run on a schedule. something like once a day, maybe once a week
+func (m *ExpirationManager) attemptIrrevocableLeasesRevoke() {
+	m.irrevocable.Range(func(k, v interface{}) bool {
+		leaseID := k.(string)
+		le := v.(*leaseEntry)
+
+		if le.ExpireTime.Add(7 * 24 * time.Hour).Before(time.Now()) {
```

Comment:
I think this could be reduced quite a bit, what about just an hour ago?

Reply:
6197b3c04